### ISSUE TITLE
Windows 11 Version 22H2 UUP Updates Fail with ConfigMgr Update Source

### ISSUE THREAD

**Issue by bbrownchg** [alaurie/WimWitchFK#4]
When using ConfigMgr as the update catalog source, the new UUP CUs from Microsoft do not get parsed properly and fail to download.

![image](https://user-images.githubusercontent.com/47155353/232856714-4a693146-f734-4b02-90be-1e9e6c92669e.png)

Based on the current code, WIM Witch uses the SourceURL attribute returned from the WMI lookup to the update catalog. The problem is, the new UUP updates have multiple source URLs and WIM Witch is not built to handle multiple source URLs being returned from the update catalog.

![image](https://user-images.githubusercontent.com/47155353/232856848-7c10f75e-bebd-459c-a235-a6f31f3ab057.png)

![image](https://user-images.githubusercontent.com/47155353/232857144-6e1c962b-0a04-4ef9-a9fd-fa9e5277a3d2.png)

Additionally, even if we were to rewrite the script to look for the update CAB file, Microsoft has [stated](https://techcommunity.microsoft.com/t5/windows-servicing/faq-wsus-and-unified-update-platform-uup-on-premises/m-p/3773235) that this CAB will not work to service a WIM and we should instead retrieve the full MSU directly from the Microsoft Update Catalog. 

![image](https://user-images.githubusercontent.com/47155353/232857767-613eca0f-8c90-4a59-bc2d-6ad13093794f.png)

David Segura has also [removed these updates from OSDSUS](https://twitter.com/SeguraOSD/status/1646356398970732544?cxt=HHwWgIC85YGbhNktAAAA) due to the same issue.

It seems like a rewrite is needed to look up the update and download it directly from Microsoft without relying on data from ConfigMgr itself.

**Comment by criiser** [comment]
AFAIK . the CM is used for WHAT to download but it is downloaded from MS directly and not from CM.. SO - IT might still work.. I will definitly look into this.. SOunds like and worthy challenge =)

**Comment by bbrownchg** [comment]
Right, it gets downloaded from Microsoft but it's grabbing the download URLs from the ConfigMgr update catalog metadata.

Per the updated documentation on the UUP updates, the ~122 MB CAB file that is part of the UUP update package (http://dl.delivery.mp.microsoft.com/filestreamingservice/files/f47a805a-18d1-48b2-9b7d-24892d58301e/public/windows11.0-kb5025239-x64_160a4310610c3eadec55ff9a7dd6ec641e411523.cab) will fail to install and we should instead be using the ~298 MB MSU directly from the Microsoft Update Catalog: https://catalog.sf.dl.delivery.mp.microsoft.com/filestreamingservice/files/9a4464d3-f890-4b6c-a3a9-e15ddbe7c9ec/public/windows11.0-kb5025239-x64_e9f2f715cfce56adc6b875512f0088aa4b3b10eb.msu.

**Comment by criiser** [comment]
AH IC ! - SO the DL URL from CM is not the valid one. Dang... Hmm, where can we get the proper one - Needs investigating... (Sigh.. Microsoft...=) )

**Comment by bbrownchg** [comment]
Haha yea, that's our lovely challenge! I wonder if the basis of this module still works? https://gist.github.com/potatoqualitee/b5ed9d584c79f4b662ec38bd63e70a2d

Though this does make me wonder if this broke in-console servicing for Windows 11 Version 22H2 since I'm pretty sure that uses DISM under the hood.

**Comment by chrisecklar** [comment]
> Haha yea, that's our lovely challenge! I wonder if the basis of this module still works? https://gist.github.com/potatoqualitee/b5ed9d584c79f4b662ec38bd63e70a2d
> 
> Though this does make me wonder if this broke in-console servicing for Windows 11 Version 22H2 since I'm pretty sure that uses DISM under the hood.

In-console servicing of Windows 11 still appears to be broken in SCCM 2203 with the latest hotfix.
